test/e2e/annotations/rewrite.go

```diff
@@ -117,4 +117,135 @@ var _ = framework.IngressNginxDescribe("Annotations - Rewrite", func() {
 		Expect(logs).To(ContainSubstring(`"(?i)/something$" matches "/something", client:`))
 		Expect(logs).To(ContainSubstring(`rewritten data: "/", args: "",`))
 	})
+
+	It("should use correct longest path match", func() {
```

Comment:
Which one of these tests is actually failing without your changes? Can we avoid introducing more tests? I think we concluded that there's only once case we are fixing, why not test just that (which is given ingress with path=/ and rewrite annotation, when I create another ingress for the same host with `/.well-known/acme/challenge` requests to `/.well-known/acme/challenge` goes to the service targeted by `/`)?

Reply:
This is the important test for that case. The other can be removed.